spec/controllers/passwords_controller_spec.rb

```diff
@@ -3,186 +3,161 @@
-          expect(@user.reload.confirmation_token).to eq @user.confirmation_token
-        end
+      it "still responds with success so as not to leak registered users" do
+        email = "this_user_does_not_exist@non_existent_domain.com"
 
-        it 'should not send a password reminder email' do
-          expect(ActionMailer::Base.deliveries).to be_empty
-        end
+        post :create, password: { email: email }
 
-        it { is_expected.to render_template(:create) }
+        expect(response).to be_success
+        expect(response).to render_template "passwords/create"
       end
     end
   end
 
-  describe 'a signed up user and forgotten password' do
-    before do
-      @user = create(:user)
-      @user.forgot_password!
-    end
+  describe "#edit" do
+    context "valid id and token are supplied" do
+      it "renders the password form for the user" do
+        user = create(:user, :with_forgotten_password)
 
-    describe 'on GET to #edit with correct id and token' do
-      before do
-        get :edit,
-            user_id: @user.to_param,
-            token: @user.confirmation_token
-      end
+        get :edit, user_id: user, token: user.confirmation_token
 
-      it 'should find the user' do
-        expect(assigns(:user)).to eq @user
+        expect(response).to be_success
+        expect(response).to render_template(:edit)
+        expect(assigns(:user)).to eq user
       end
-
-      it { is_expected.to respond_with(:success) }
-      it { is_expected.to render_template(:edit) }
     end
 
-    describe 'on GET to #edit with correct id but blank token' do
-      before do
-        get :edit, user_id: @user.to_param, token: ''
-      end
-
-      it { is_expected.to set_the_flash.to(/double check the URL/i).now }
-      it { is_expected.to render_template(:new) }
-    end
+    context "blank token is supplied" do
+      it "renders the new password reset form with a flash notice" do
+        get :edit, user_id: 1, token: ''
 
-    describe 'on GET to #edit with correct id but no token' do
-      before do
-        get :edit, user_id: @user.to_param
+        expect(response).to render_template(:new)
+        expect(flash.now[:notice]).to match(/double check the URL/i)
       end
-
-      it { is_expected.to set_the_flash.to(/double check the URL/i).now }
-      it { is_expected.to render_template(:new) }
     end
 
-    describe 'on PUT to #update with password' do
-      before do
-        @new_password = 'new_password'
-        @old_encrypted_password = @user.encrypted_password
+    context "invalid token is supplied" do
+      it "renders the new password reset form with a flash notice" do
+        user = create(:user, :with_forgotten_password)
 
-        put :update, user_id: @user, token: @user.confirmation_token,
-          password_reset: { password: @new_password }
-        @user.reload
-      end
+        get :edit, user_id: 1, token: user.confirmation_token + 'a'
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.